service/SilberService.cpp

```diff
@@ -0,0 +1,112 @@
+                std::cout << "received SET_CONFIG req px:" << messageConf->configuration.MinPixelsForObject << std::endl;
+                Message msg{message->id, MessageType::SET_CONFIG_OK};
+                auto iter = m_connections.processActionUpdate(message);
+                // Configuration newConf{messageConf->configuration.AffinityThreshold, messageConf->configuration.MinPixelsForObject, messageConf->configuration.PixelStep, messageConf->configuration.CalculationTimeLimit, messageConf->configuration.IdleTimeout, messageConf->configuration.ThreadsMultiplier};
+
+                m_amApi->setConfiguration(messageConf->configuration);
+                std::cout << "received SET_CONFIG OK \n";
+                m_server->send(&msg);
+                std::cout << "received SET_CONFIG OK sent\n";
+            }
+            else if (message->type == MessageType::COMPARE_REQUEST)
+            {
+                std::cout << "received COMPARE_REQUEST req\n";
+                auto iter = m_connections.processActionUpdate(message);
+                m_amApi->setConfiguration(iter->configuration);
+
+                MessageCompareRequest *messageCompare = static_cast<MessageCompareRequest *>(message);
+                if (messageCompare)
+                {
+                    printf("compare_ %s %s _\n", messageCompare->base, messageCompare->to_compare);
+                    auto iter = m_connections.processActionUpdate(message);
+                    m_amApi->setConfiguration(iter->configuration); // set configuration for this client
+                    printf("conf pixs %zu\n", iter->configuration.MinPixelsForObject);
+                    am::analyze::algorithm::DescObjects result;
+                    try
+                    {
+                        result = m_amApi->compare(messageCompare->base, messageCompare->to_compare);
+                    }
+                    catch (const am::common::exceptions::AmException exc)
+                    {
+                        std::cout << "Exception has been caught: " << exc.what() << ::std::endl;
+                        Message failed{messageCompare->id, MessageType::COMPARE_FAIL};
+                        m_server->send(&failed);
+                        continue;
+                    }
+                    std::cout << "received after compare\n";
+                    MessageCompareResult compare_result;
+                    compare_result.id = messageCompare->id;
+
+                    compare_result.payload_bytes = result.size() * sizeof(Rect);
+                    Rect *rects = static_cast<Rect *>(compare_result.payload);
+                    size_t id = 0;
+                    for (auto &rectangle : result)
+                    {
+                        if (id == 100) // limit by 100 objects
+                        {
+                            compare_result.payload_bytes = 100 * sizeof(Rect);
+                            break;
+                        }
```

Comment:
All possible exceptions have to be handled.
An instance of a shared memory have object has to be destroyed to be able to reuse synchronization primitives.
In case of forced termination of aquamarine process such sync primitives could be left in an unknown state and work with mapped memory have to be finalized correctly.